private-templates-service/adaptivecards-templating-service/src/TemplateServiceClient.ts

```diff
@@ -376,43 +376,76 @@ export class TemplateServiceClient {
     let existingTemplate: ITemplate = response.result[0];
     let templateName: string = name ? name : existingTemplate.name;
     let existingTags = existingTemplate.tags;
+    
     if (tag) {
       existingTags!.push(tag);
     }
     let tags: string[] | undefined = tag ? existingTags : tagList;
-    let templateState: TemplateState | undefined = isPublished ? TemplateState.live : state;
+    let templateState: TemplateState | undefined =  state;
 
+    if(!version){
+      version = incrementVersion(existingTemplate);
+    }
+ 
     let templateInstance: ITemplateInstance = {
       json: template ? template : JSON.parse("{}"),
       version: version || "1.0",
       state: templateState,
       data: [],
-      publishedAt: isPublished ? new Date(Date.now()) : undefined,
+      publishedAt: state === TemplateState.live ? new Date(Date.now()) : undefined,
       updatedAt: new Date(Date.now()),
       numHits: 0,
       isShareable: isShareable
     };
-
     let templateInstances: ITemplateInstance[] = [];
     if (existingTemplate.instances) {
       // Check if updated version already exists
       let added = false;
       for (let instance of existingTemplate.instances) {
-        if (instance.version === templateInstance.version) {
+        if (instance.version === templateInstance.version) { 
+          
+          if(instance.state === TemplateState.deprecated){
+            // the template that is trying to be modified is deprecated
```

Comment:
personal preference - make sure comments that get committed follow grammar/punctuation - makes them look more professional, rather than scribbled notes. Here - I'd use capitalize the first word.